### ISSUE TITLE
Transaction time / charge time

### ISSUE THREAD

**Issue by woj-tek** [mtotschnig/MyExpenses#421]
This applies mostly to card transactions where we have two dates: when the transaction was made and the second - when it was charged from our account.
Currently we could adjust the time after "clearing" the account but (at least I) would like to maintain original date of the transaction (leaving it with the original date causes skew with the bank statement which is inconvenient)

**Comment by mtotschnig** [comment]
@woj-tek I am wondering about the terms you have suggested. Would they be equivalent with 
"booking date" and "value date"? In German bank statements, it would read as "Buchungstag" und "Wertstellung".

**Comment by woj-tek** [comment]
My dictionary says that the first one (transaction time) would be Buchungstag and charge time would be Wertstellung.

Basically one date is when you actually used card (quite often after that the amount paid is under "locked transactions" and then the shop actually charges the card (takes the amount from your account) and this is the charge time Wertstellung (change of the actual Saldo of your account)? Before charging the account you could see that your Saldo is x, but available amount of money is Saldo-<locked transactions sum>.

**Comment by mtotschnig** [comment]
I am planning to package this with #400 together into next release. This brought up the following question: Since the use case for this ticket are dates as handled by your bank, which normally do not carry time information, should we also limit the implementation to deal with dates only?

**Comment by woj-tek** [comment]
Yeah, makes sense. (and with linked ticket in mind those two date fields could nicely go side-by-side instead of having time filed :-) )

**Comment by mtotschnig** [comment]
@woj-tek Thanks for the valuable feedback. One final clarification: Which of the two dates should be taken into account for grouping.

**Comment by mtotschnig** [comment]
One other complication I discovered during work on this ticket: If I am not mistaken for transfers the value date could be different for the two parts of the transfer: I.e when you transfer from one bank account to the other, it could be charged on an earlier date on the former, than credited on the latter. Currently the UI for transfers does not easily allow to take this into account. I am undecided if this should be tackled now, or left for a later optimization. @woj-tek Any thoughts?

**Comment by mtotschnig** [comment]
For transfers the value date is now recorded separately for each side of the transfer, i.e. depending on from which account it is edited.

**Comment by woj-tek** [comment]
Yes, this is correct that for transfer the dates may be different, but (at least for me) most of the time they occur on the same day so I haven't run into it yet.
Thanks for taking care of it :-)
